ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -1557,142 +1537,80 @@ def test031_user_can_add_suborder_with_test_units(self):
+        suborders, _ = self.orders_api.get_suborder_by_order_id(order['id'])
+        suborder = suborders['orders'][0]
+        suborder_update_index = len(suborders['orders']) - 1
+        test_units = [test_unit['testUnit']['name'] for test_unit in suborder['testUnit']]
+
+        self.info('get random completed test plan with different article')
+        test_plans = TestPlanAPI().get_completed_testplans()
+        test_plans_without_duplicate = [test_plan for test_plan in test_plans if
+                                        test_plan['materialType'] == suborder['materialType'] and
+                                        suborder['article'] != test_plan['article'][0]]
+        if test_plans_without_duplicate:
+            test_plan_data = random.choice(test_plans_without_duplicate)
+            test_plan = test_plan_data['testPlanName']
+            article = test_plan_data['article'][0]
+        else:
+            article = ArticleAPI().get_aticle_with_material_type(suborder['materialType'])
+            new_test_plan = TestPlanAPI().create_completed_testplan(
+                material_type=suborder['materialType'], article=article)
+            test_plan = new_test_plan['testPlanEntity']['name']
+
+        self.info('update order {} with article {}'.format(order['orderNo'], article))
+        self.orders_page.get_order_edit_page_by_id(order['id'])
+        if article == 'all':
+            self.order_page.update_suborder(sub_order_index=suborder_update_index, articles='')
+            article = self.order_page.get_article()
+        else:
+            self.order_page.update_suborder(sub_order_index=suborder_update_index, articles=article)
 
-    #         child_table = self.analyses_page.get_child_table_data(index=0)
-    #         analysis_testunits = []
-    #         for record in child_table:
-    #             analysis_testunits.append(record['Test Unit'])
+        self.info('assert test plan and articles are empty')
+        self.orders_page.sleep_tiny()
+        self.assertFalse(self.order_page.get_test_plan())
+        if test_units:
+            self.assertCountEqual([self.order_page.get_test_unit()], test_units)
+        else:
+            self.assertFalse(self.order_page.get_test_unit())
 
-    #         self.base_selenium.LOGGER.info('+ Assert analysis test units are: {}, and it should be: {}'.format(analysis_testunits, suborder_testunits_after_refresh))
-    #         self.assertEqual(set(suborder_testunits_after_refresh) == set(analysis_testunits), True)
+        if save == 'save':
+            self.order_page.set_test_plan(test_plan)
+            self.info('save the changes then refresh')
+            self.order_page.save(save_btn='order:save_btn')
+            self.order_page.get_orders_page()
+        else:
+            self.order_page.cancel()
+            self.order_page.confirm_popup()
+            article = suborder['article']
+            test_plan = suborder['testPlans'][0]
```

Comment:
after cancel you press on the cancel button you need to assert that the article didn't change

Reply:
I can't make assertion here as cancel button redirect me to order page